### DISCUSSION TITLE
Apache as ReverseProxy for ldproxy

### DISCUSSION THREAD

**Discussion by mattenp** [ldproxy/ldproxy#1051] [Setup] [answered]
Hi,

I would like to use Apache Web Server as reverse proxy for ldproxy, but it does not really works.

Apache config:
```
ProxyPass /ldproxy http://localhost:7080
ProxyPassReverse /ldproxy http://localhost:7080
```

ldproxy as extracted distTar
data/cfg.yml
```
server:
  externalUrl: https://hostname/ldproxy
```

Best regards,

Matthias

Federal Maritime and Hydrographic Agency Hamburg

**Comment by cportele** [comment, marked as answer]
From the description it is unclear what "does not work" means, i.e., what the errors are, what is logged, etc. It is not an ldproxy problem, is is a problem in your Apache configuration.

Possible reasons include:

- `mod_proxy` or `mod_proxy_http` are not loaded
- some error in the Apache https configuration (since you are using https in the external URL)

Aside from that I would recommend to redirect as follows to block access to the admin resources for external access:

```
ProxyPass /ldproxy http://localhost:7080/rest/services
ProxyPassReverse /ldproxy http://localhost:7080/rest/services
```

**Reply by mattenp** [reply]
Thanks a lot Clemens
